File: tests/fixtures/rules-tests/knockout/knockout.py
# KNOCKOUT: completely unrelated code — no LLM, no auth, no security context
# This file should NOT trigger any security rule.
def calculate_fibonacci(n: int) -> list[int]:
    """Generate Fibonacci sequence up to n terms."""
    if n <= 0:
        return []
    if n == 1:
        return [0]
    seq = [0, 1]
    for i in range(2, n):
        seq.append(seq[i - 1] + seq[i - 2])
    return seq


def format_temperature(celsius: float) -> str:
    """Convert Celsius to Fahrenheit and format."""
    fahrenheit = (celsius * 9 / 5) + 32
    return f"{celsius}°C = {fahrenheit:.1f}°F"


def count_words(text: str) -> dict[str, int]:
    """Count word frequencies in text."""
    counts = {}
    for word in text.lower().split():
        word = word.strip(".,!?;:'\"()[]{}")
        if word:
            counts[word] = counts.get(word, 0) + 1
    return counts


def is_palindrome(s: str) -> bool:
    """Check if string is a palindrome."""
    cleaned = "".join(c.lower() for c in s if c.isalnum())
    return cleaned == cleaned[::-1]


def merge_sorted_lists(a: list[int], b: list[int]) -> list[int]:
    """Merge two sorted lists."""
    result = []
    i = j = 0
    while i < len(a) and j < len(b):
        if a[i] < b[j]:
            result.append(a[i])
            i += 1
        else:
            result.append(b[j])
            j += 1
    result.extend(a[i:])
    result.extend(b[j:])
    return result

File: tests/fixtures/rules-tests/knockout/knockout.dart
// KNOCKOUT: completely unrelated code — no badCertificateCallback, no security
// This file should NOT trigger any security rule.

int factorial(int n) {
  if (n <= 1) return 1;
  return n * factorial(n - 1);
}

double calculateArea(double radius) {
  return 3.14159 * radius * radius;
}

List<int> fibonacci(int n) {
  if (n <= 0) return [];
  if (n == 1) return [0];
  final list = [0, 1];
  for (int i = 2; i < n; i++) {
    list.add(list[i - 1] + list[i - 2]);
  }
  return list;
}

Map<String, int> countWords(String text) {
  final counts = <String, int>{};
  for (final word in text.toLowerCase().split(' ')) {
    counts[word] = (counts[word] ?? 0) + 1;
  }
  return counts;
}
